### PR TITLE
LLVM 12.0.1

### PR DESCRIPTION
**Pull request by liushuyu**
Topic Description
-----------------

Update LLVM to 12.0.1

Package(s) Affected
-------------------

```
llvm
opencl-clang
intel-graphics-compiler
ispc
mesa
codelite
intellij-idea
bpftrace
opencl-registry-api
level-zero
intel-gmmlib
metee
igsc
intel-compute-runtime
```

Security Update?
----------------

No

Architectural Progress
----------------------

- [X] AMD64 `amd64`   
- [X] AArch64 `arm64`

Secondary Architectural Progress
--------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

Post-Merge Secondary Architectural Progress
-------------------------------------------

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

### CODE REVIEW

**Comment by liushuyu** [comment]
Supersedes #3269.

File: extra-devel/opencl-registry-api/autobuild/defines [hunk 1]
```diff
@@ -1,7 +1,8 @@
 PKGNAME=opencl-registry-api
 PKGSEC=devel
 PKGSUG="libcl"
-BUILDDEP="curl"
+BUILDDEP="curl cmake"
```
Comment (MingcongBai) [inline extra-devel/opencl-registry-api/autobuild/defines:4]: `cmake` is preinstalled with BuildKit.

File: extra-libs/igsc/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,5 @@
+PKGNAME=igsc
+PKGSEC=libs
+PKGDEP="gcc-runtime"
+BUILDDEP="cmake ninja metee"
```
Comment (MingcongBai) [inline extra-libs/igsc/autobuild/defines:4]: Ditto above.

File: extra-libs/metee/autobuild/defines [hunk 3]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=metee
+PKGSEC=libs
+PKGDEP="gcc-runtime"
+BUILDDEP="cmake ninja"
```
Comment (MingcongBai) [inline extra-libs/metee/autobuild/defines:4]: Ditto above.

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**